frontend/src/features/admin-form/create/workflow/components/WorkflowContent/EditStepBlock/RespondentBlock.tsx

```diff
@@ -170,19 +204,426 @@ const DynamicRespondentOption = ({
+  const selectedConditionalFieldOptionsToRecipientsMap =
+    selectedConditionalField?.optionsToRecipientsMap
+
+  const isOptionsToRecipientsMapAttached = !!csvFile
+
+  const standardCsvDownloadFileName = `conditional_routing_form_${formId}_field_${selectedConditionalFieldTitle}_mapping.csv`
+
+  const placeholderOptionToEmailMappingCsv = useMemo(() => {
+    return {
+      name: standardCsvDownloadFileName,
+      type: 'text/csv',
+    } as File
+  }, [standardCsvDownloadFileName])
+
+  useEffect(() => {
+    if (selectedConditionalFieldOptionsToRecipientsMap) {
+      setCsvFile(placeholderOptionToEmailMappingCsv)
+    } else {
+      setCsvFile(null)
+    }
+  }, [
+    placeholderOptionToEmailMappingCsv,
+    setCsvFile,
+    selectedConditionalFieldOptionsToRecipientsMap,
+  ])
+
+  const { isOpen, onOpen, onClose } = useDisclosure()
+
+  const handleCsvDownload = () => {
+    if (!selectedConditionalFieldOptionsToRecipientsMap) return
+    const csvData = {
+      fields: ['Options', 'Add email(s) in this column'],
+      data: Object.entries(selectedConditionalFieldOptionsToRecipientsMap).map(
+        ([option, recipients]) => [option, recipients.join(',')],
+      ),
+    }
+
+    const csvString = Papa.unparse(csvData, {
+      header: true,
+      delimiter: ',',
+      skipEmptyLines: 'greedy',
+    })
+    const csvBlob = new Blob([csvString], {
+      type: 'text/csv',
+    })
+    const csvFile = new File([csvBlob], standardCsvDownloadFileName, {
+      type: 'text/csv',
+    })
+    downloadFile(csvFile)
+  }
+
+  const handleSkeletonCsvDownload =
+    (formId: string = '') =>
+    () => {
+      const getFieldOptions = (conditionalFieldId: string) => {
+        const conditionalField = conditionalFormFields.find(
+          (field) => field._id === conditionalFieldId,
+        )
+        return conditionalField?.fieldOptions
+      }
+      const generateCsvContent = (fieldOptions: string[] | undefined) => {
+        const headerRow = ['Options', 'Add email(s) in this column']
+        const optionsRows = fieldOptions?.map((field) => [field, '']) ?? []
+        const jsonContent = [headerRow, ...optionsRows]
+        return Papa.unparse(jsonContent, {
+          header: true,
+          delimiter: ',',
+        })
+      }
+
+      const csvStringToFile = (csvString: string, downloadFileName: string) => {
+        const csvBlob = new Blob([csvString], {
+          type: 'text/csv',
+        })
+        const csvFile = new File([csvBlob], downloadFileName, {
+          type: 'text/csv',
+        })
+        return csvFile
+      }
+
+      const conditionalFieldId = watch('conditional_field')
+      if (!conditionalFieldId) return
+
+      const fieldOptions = getFieldOptions(conditionalFieldId)
+      const csvContent = generateCsvContent(fieldOptions)
+      const csvFile = csvStringToFile(
+        csvContent,
+        `conditional_routing_form_${formId}_field_${selectedConditionalFieldTitle}_mapping.csv`,
```

Comment:
This seemed to be re-used above. Understand that we likely want to avoid the stale closure problems thus have it this way. Another approach to reuse could be to convert this into a function instead.


```tsx
function getFileName(formId: string, fieldTitle: string) {
  return `conditional_routing_form_${formId}_field_${fieldTitle}_mapping.csv`
}

const ConditionalRoutingOption = ({...}) => {
  const standardCsvDownloadFileName = getFileName(formId, selectedConditionalFieldTitle)
  ...
  const handleSkeletonCsvDownload = (formId, ...) => { 
      ...
      const csvFile = csvStringToFile(
        csvContent,
        getFileName(formId, selectedConditionalFieldTitle)
      )
  }
}
```

Reply:
resolved in [615b39a](https://github.com/opengovsg/FormSG/pull/7804/commits/615b39ab6eac0368091e037be7d66405004fe88a)